src/toolkits/drawing_classifier/data_preparation.cpp

```diff
@@ -230,84 +231,132 @@ flex_list simplify_drawing(flex_list raw_drawing, int row_number) {
     return simplified_drawing;
 }
 
-bool in_bounds(int x, int y, int dim) {
-    return (x >= 0 && x < dim && y >= 0 && y < dim);
+bool in_bounds(int x, int y, int x_dim, int y_dim) {
+    return (x >= 0 && x < x_dim && y >= 0 && y < y_dim);
 }
 
 void paint_point(flex_nd_vec &bitmap, int x, int y, int pad) {
-    size_t dimension = bitmap.shape()[1];
+    size_t x_dimension = bitmap.shape()[0];
+    size_t y_dimension = bitmap.shape()[1];
     for (int dx = -pad; dx < pad; dx++) {
         for (int dy = -pad; dy < pad; dy++) {
-            if (in_bounds(x+dx, y+dy, dimension)) {
-                bitmap[(y+dy) * dimension + (x+dx)] = 1.0;
+            if (in_bounds(x + dx, y + dy, x_dimension, y_dimension)) {
+                DASSERT_GE(x + dx, 0);
+                DASSERT_LT(x + dx, x_dimension);
+                DASSERT_GE(y + dy, 0);
+                DASSERT_LT(y + dy, y_dimension);
+#ifndef NDEBUG
```

Comment:
@hoytak Do we need an #ifndef NDEBUG ... #endif here because `indices` might be an unused variable in `release`?

Reply:
Yes, likely.